### PR TITLE
Configure environment for SSM Session Manager

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

This pull request configures the environment for use with SSM Session Manager.

## 💭 Motivation and Context

SSM Session Manager allows privileged users to connect to EC2 instances via the AWS control plane using `ssh`, `aws ssm start-session`, or the AWS Console.  This means you can get a shell into any EC2 instance, even if port 22 is not open or if the instance is completely isolated without any access to an internet Gateway!

This is obviously a boon for security, and it allowed me to remove some port 22 ACL and Security Group rules from this Terraform code.

The key to starting sessions is to assume the `StartStopSSMSession` role, which the dev users are allowed to do.  You can start sessions:
* [Via the AWS Console.](https://console.aws.amazon.com/systems-manager/session-manager/start-session?region=us-east-1) (Here you can login as the usual administrative user we normally use.)
* Via `aws --profile cool-env0-startstopssmsession --region us-east-1 ssm start-session --target i-XXXXXXXXXXXXXXXXX --document-name SSM-SessionManagerRunShell`, after installing [the `session-manager-plugin`](https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html#install-plugin-macos).
* Via `AWS_PROFILE=cool-env0-startstopssmsession AWS_DEFAULT_REGION=us-east-1 ssh i-XXXXXXXXXXXXXXXXX`, after [adding a `ProxyCommand` to your ssh configuration](https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-getting-started-enable-ssh-connections.html) and installing [the `session-manager-plugin`](https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-install-plugin.html#install-plugin-macos).
* [_And there's more!_](https://docs.aws.amazon.com/systems-manager/latest/userguide/session-manager-working-with-sessions-start.html)

## 🧪 Testing

I have deployed these changes to env0-staging and verified that I can connect to each of the instances there via SSM Session Manager in all three ways.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: nessus_iam.tf [hunk 1]
```diff
@@ -36,6 +36,15 @@ resource "aws_iam_role_policy_attachment" "cloudwatch_agent_policy_attachment_ne
...
 
+# Attach the SSM Agent policy to this role as well
+resource "aws_iam_role_policy_attachment" "ssm_agent_policy_attachment_nessus" {
+  count    = lookup(var.operations_instance_counts, "nessus", 0) > 0 ? 1 : 0
+  provider = aws.provisionassessment
+
+  role       = aws_iam_role.nessus_instance_role[count.index].id
+  policy_arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+}
+
```
Comment (dav3r) [inline nessus_iam.tf:47]: I just merged the [PR that adds `gophish_iam.tf`](https://github.com/cisagov/cool-assessment-terraform/pull/45) - can you please add this sauce to that file as well?

**Review by dav3r — APPROVED** [review]
This looks fantastic!  💪 💪 💪 💼 🧠

File: nessus_iam.tf (hunk 1 above)
Comment (jsf9k) [reply nessus_iam.tf:47]: Will do!

Comment (jsf9k) [reply nessus_iam.tf:47]: See commit 64bff80 and cisagov/pca-gophish-composition-packer#8.

**Review by felddy — APPROVED** [review]
Whoa!  This is pretty cool.  I think we'll need a show-n-tell tomorrow!  
🌟 🍪 🌟

**Pull request merged**